### PR TITLE
[significant] database + event system + area system completed

### PR DESCRIPTION
**Pull request by mahirhassan-uoft**
- implemented database functionality & fixed issues
- implemented area graph-like system with adjacent areas:
    - constructor initializes area data manager automatically
    - also initializes introduction to begin the game
    - generates area graph and sets current area
    - added functionality to move between areas and also switch zones (getToNextArea)
- implemented events and executable event formatting
    - allows for queueing of events
    - capable of automatically executing events upon entering areas
    - multiple types of events have been created, but they must be implemented:
        - juhwan: ItemPickUpEvent
        - marja-leena/yasu/charlie: EncounterEvent
- added tests for all components and thoroughly tested all managers
- added docstrings to help understand functionality

please refer to tests for examples of how to use each manager
i will also extend the database manager for everyones systems later on

### CODE REVIEW

File: src/main/resources/Database.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      "To Your Demise": {
+
+      }
+    }
+  },
+  "quests": [],
+  "characters": [],
+  "items": [],
+  "skills": [],
+  "enemies": [],
```
Comment (Yasutakahi) [inline src/main/resources/Database.json:103]: For now it is completely fine, but I would like 2 other sections for bosses and enemyai. Bosses basically stores the same information as enemy but stores additional information which is gimmick. Enemyai stores the name of the AI which are default or smart, and also stores integer called chance (chance represents the chance of attack for an enemy).

**Review by Yasutakahi — DISMISSED** [review]
Although there are some additional information that I would like you to add, I understand that it will be added later. All the other things like managers, it looks very nice in terms of clean architecture and solid principle. However, I found some packaging issues which is package for unit test. I think it looks nicer if tests are also packaged like other classes.

**Review by Yasutakahi — APPROVED** [review]
Now the database of the enemy, boss, skill, and ai are added which have all the information the enemy system needs. Moreover, other components such as clean architecture and solid looks very nice to me. So I think overall it is very well coded and organized.

**Review by charlietaoxl — APPROVED** [review]
So far the implementation of the AreaManager and EventManager are quite clear. However, for EventManager and Events, are events in this situation supposed to have the functionality of States? 

In the case where all of this code will be executed by the Shell, the shell will be running through the state managers and the state managers are what decide which state should be executed next. Maybe implementing State for Event and making an EventStateManager would be cleaner for the Shell to execute, since most likely these will be running in a StateManager execution loop in the Shell. 

However you want to integrate States and StateManagers into Events is up to you, unless there is another method you thought of for Events to be executed. For now since there is not much implementation of the Events and such I would say it's fine to merge for now for dependency purposes.

**Review by juhwansonn — APPROVED** [review]
Implementation follows clean architecture and SOLID principle.
I think you are missing javadocs for some methods and attributes, maybe you considered those attributes and methods minor things. But I think it's good practice to add javadocs to all attributes and methods even if they are not important. 
Although we should talk about item database later, I think you can merge them for now.

File: src/main/resources/Database.json [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+      },
+      "To Your Demise": {
+
+      }
+    }
+  },
+  "quests": [],
+  "characters": [],
+  "items": [],
```
Comment (juhwansonn) [inline src/main/resources/Database.json:101]: you can add "price", "name", "ability", "level" to items

**Pull request merged**